### DISCUSSION TITLE
Newbie to Macro - Partial success only. Switch PTZ to Media and back again.

### DISCUSSION THREAD

**Discussion by Chipperchap** [WarmUpTill/SceneSwitcher#767] [Q&A] [answered]
We send out a PTZ camera source via the "Virtual Camera" to ZOOM. However at our venue we sometimes show media (not generated by OBS) on auditorium screens, and I would like the participants on ZOOM to automatically view the media being displayed without having us having to use the  ZOOM Share Screen. 

In OBS I have scenes for:

- Media (content from another piece of software shown on a screen/monitor on the auditorium wall)

- PTZ camera

I am seeking a macro to automatically switch to display for ZOOM the image from our PTZ when it detects that a screen/monitor is in a standby mode / no new media is being displayed. (Incidentally on that screen we have some text as a PNG file being displayed by another piece of software, not generated by OBS.) 
Then when new media is displayed by that other software on the screen/monitor, I wish OBS it to switch automatically from our PTZ image to the displayed media to send it out to the virtual camera (for ZOOM).
Once the media is finished and the screens return to the static image, I would really like it if OBS would detect that and switch back to our PTZ camera image again. 

I can get OBS to switch from the displayed static image on the auditorium screens to the PTZ camera, but when the screen changes to display media, I cannot get the PTZ scene to deactivate to now show the media on the auditorium screen. 

Any help please would be greatly appreciated.

**Comment by WarmUpTill** [comment]
I would assume that using the [Video condition](https://github.com/WarmUpTill/SceneSwitcher/wiki/Video-condition) could work here.

You could set up two macros to detect if the media scene is changing and switch scenes accordingly.
Here is an example:

![Video1](https://github.com/WarmUpTill/SceneSwitcher/assets/19472752/ee55d1bc-aaca-4e21-acf9-ce5c1c006bde)
![Video2](https://github.com/WarmUpTill/SceneSwitcher/assets/19472752/06cb0f1d-5715-454e-8dc7-260b5aeb0197)

**Reply by Chipperchap** [reply]
Awesome - we are now getting closer to the solution. 

I followed your screenshot suggestions exactly except changed it to "for at least" one seconds.
This is however the behaviour I am getting...
 
https://github.com/WarmUpTill/SceneSwitcher/assets/35364657/a3bc60fa-20a2-483b-a11b-3ac1ec5f9e1c

Thank you so much for any further suggestions.

**Reply by WarmUpTill** [reply]
Unfortunately it is a bit difficult to say without having more details on what is being displayed on the scenes at the moment of the scene switch.

This _might_ happen because of the removal of the duration modifier, as now even the slightest change in the media scene will trigger a scene switch.

You can maybe try to enable pattern matching with a very high similarity threshold to ignore to ignore these minor changes.

Let me know if this helps! :)

**Comment by Chipperchap** [comment]
Looks like I am having a bit more success.
It still oscillates but at least now when I show different media it will stop and show the media.
For this exercise I am just using a nothing .png and photo gallery as the media player. If I can get these basic things to work hopefully my other software will too... (It is 3am in morning so I don't have any lights on in my office so camera is dark 😎)
Please see screenshot video.

https://github.com/WarmUpTill/SceneSwitcher/assets/35364657/2ee2b7e6-c8f7-4b37-ade3-2a28811ca97b

Here are my settings.

![Screenshot 2023-06-05 at 3 23 26 AM](https://github.com/WarmUpTill/SceneSwitcher/assets/35364657/d96bf991-a99b-4f72-9606-3acdb36b3bf4)

![Screenshot 2023-06-05 at 3 23 55 AM](https://github.com/WarmUpTill/SceneSwitcher/assets/35364657/ce2fbed2-e7d5-47a1-bb07-8e3bc80a9b91)

**Reply by WarmUpTill** [reply]
I think the issue is in my initial example. (Can't check to confirm myself unfortunately)

Can you please check if unchecking the "perform action only on condition change" in the "Media stopped" macro makes difference?

**Comment by Chipperchap** [comment]
Thank you so much for your continued assistance. 

I regret to report that unchecking that box makes no difference. 
I tried also without a "duration modifier" but that also made no change.  
Video below is showing the continual oscillation between the scenes. 

https://github.com/WarmUpTill/SceneSwitcher/assets/35364657/b56e8fc7-88c3-4e0c-a96e-f37ccaa9b8f9

It almost seems like there is a conflict happening somewhere but as I have no "Macro" experience I am likely to be wrong.

**Reply by WarmUpTill** [reply]
Can you share an OBS log with verbose logging enabled?

![](https://user-images.githubusercontent.com/19472752/239974352-adee59e2-71b9-4cd9-9c36-b16984a38fd6.PNG)

This might give us further hints why this is happening.

**Reply by Chipperchap** [reply]
Is this what you mean?: https://obsproject.com/logs/WW4kZQo9tqTNUitJ

**Reply by WarmUpTill** [reply]
My next guess would be that the pattern matching is somehow functioning as expected as soon as the scene switch happens as the scene and its sources where the pattern matching should take place are no longer being rendered.

Can you check if opening a "projector" window of the given scene / sources makes a difference?
https://github.com/WarmUpTill/SceneSwitcher/wiki/Video-condition#important-note

**Reply by Chipperchap** [reply]
I can confirm that if I open a "projector" window then the switch from standby picture to PTZ is stable. When I put up another picture it flashes to that picture for a moment then switches back to PTZ which is not the desired request. 


https://github.com/WarmUpTill/SceneSwitcher/assets/35364657/dde220da-854b-47ff-baf2-e393a0d93e3e

**Reply by WarmUpTill** [reply]
Can you please attach another log file?

**Reply by Chipperchap** [reply]
This is a log file with the Windowed Projector:

https://obsproject.com/logs/kytkyBRN1xOlOGMx

**Reply by WarmUpTill** [reply]
As the video condition only returns `false` for a single interval, I think the issue is with the pattern match settings.
Without having access to the images it is very difficult to recommend how to proceed.

I would _assume_ that the matching threshold might be to generous - but that is just a guess.

If you don't mind let's maybe set up a call where you share your screen so we can try to figure out the correct settings "live" and speed up the communication.

Feel free to contact me on the OBS forums / OBS Discord to set this up.

**Reply by Chipperchap** [reply]
I think I will just resort to using HotKeys to switch between scenes as it will be robust and controlled. 
I was sincerely hoping to automate to make it easier for other users of our A/V desk, but it appears to be much too difficult to set up. I cant see "matching thresholds" being so sensitive working for us, as we cannot have unexpected or random switching's when a speaker (talker) is describing what is being shown on the screens only to have them disappear. 
😣😢
As for a 'live' session - trouble is I am in (likely) another hemisphere to most OBS folk so when you are awake I am asleep. (I am in GMT +12)

**Reply by WarmUpTill** [reply]
I understand.
If manually switching scenes is an option it might indeed be the better save choice then. :)

>I cant see "matching thresholds" being so sensitive working for us, as we cannot have unexpected or random switching's when a speaker (talker) is describing what is being shown on the screens only to have them disappear.

Just to clarify.
The current issue is likely that the pattern matching is _too_ sensitive and it should be stricter.
So the pattern matching slider should be moved more to the right. (Probably much more as it is a static image that will not change)
In the current logs the video condition almost always returns "1" thus indicating a pattern match.

>As for a 'live' session - trouble is I am in (likely) another hemisphere to most OBS folk so when you are awake I am asleep. (I am in GMT +12)

I am in GMT +2 so that might indeed be tricky.
But I am usually quite flexible on the weekend and don't mind waking up early / staying up late to get this sorted out so the offer still stands :)

**Comment by Chipperchap** [comment]
So it take it GMT +2 is 10 hours behind me. So if I got up at 7:00am that would be 9:00pm for you. I can do that tomorrow (Saturday for me) or Sunday (for me).
Not sure of the logistics... I created an account with OBS Forums if that helps...
I am also thoroughly familiar with ZOOM. 
😀

**Reply by WarmUpTill** [reply]
That works for me - I sent you a message on the OBS forums!

**Comment by Chipperchap** [comment]
Done - Message sent 😁

**Comment by Chipperchap** [comment, marked as answer]
SOLVED!

How?
- Create macro for windowed projector to auto start. Manually minimize once started. 
- Use "Matches Pattern" in "Scene" settings with high threshold. 

Thank you so much @WarmUpTill - really appreciated!  Really appreciate your generous offer of your time.
This is going to make things so much easier for all the A/V team. 
🎁😀
